client/components/DataManagement/DataManagementRow/index.jsx

```diff
@@ -425,12 +425,53 @@ const DataManagementRow = ({
             return otherVersionsInProgress.length;
         };
 
-        if (recommendedTestPlanVersions.length) {
-            const { earliestVersion, earliestVersionDate } = getVersionData(
-                recommendedTestPlanVersions,
+        /**
+         * Uses the truthy state of defined dates for a TestPlanVersion to
+         * determine if the version was ever in that phase at any given point.
+         * Useful if the phase of the TestPlanVersion is currently DEPRECATED.
+         * @param {"RECOMMENDED"|"CANDIDATE"|"DRAFT"} phase
+         * @returns {TestPlanVersion[]}
+         */
+        const deriveVersionsByDates = phase => {
+            switch (phase) {
+                case 'RECOMMENDED':
+                    return testPlanVersions.filter(
+                        ({ recommendedPhaseReachedAt }) =>
+                            !!recommendedPhaseReachedAt
+                    );
+                case 'CANDIDATE':
+                    return testPlanVersions.filter(
+                        ({
+                            candidatePhaseReachedAt,
+                            recommendedPhaseReachedAt
+                        }) =>
+                            !!candidatePhaseReachedAt &&
+                            !recommendedPhaseReachedAt
+                    );
+                case 'DRAFT':
+                    return testPlanVersions.filter(
+                        ({
+                            draftPhaseReachedAt,
+                            candidatePhaseReachedAt,
+                            recommendedPhaseReachedAt
+                        }) =>
+                            !!draftPhaseReachedAt &&
+                            !candidatePhaseReachedAt &&
+                            !recommendedPhaseReachedAt
+                    );
+            }
+        };
+
+        const derivedRecommendedVersions = deriveVersionsByDates('RECOMMENDED');
+        const derivedCandidateVersions = deriveVersionsByDates('CANDIDATE');
+        const derivedDraftVersions = deriveVersionsByDates('DRAFT');
```

Comment:
Just an idea to call this `versionsThatReachedDraft`? And `versionsThatReachedRecommended`? When I read the code with the word "derived" I originally felt unsure about why this value needed to be derived and I didn't get the connection to the deprecated phase.

Likewise, for `deriveVersionsByDates` could we call that `getVersionsThatReachedPhase`?

Reply:
Sure, +1 for the consistency. 'Reached' is the verb that's currently being used in that context.